File: mern_stack_python_7842/client/src/components/UserBlogs.py
import React from 'react'
from flask import Flask, jsonify, request
import requests
from components.Blogs import Blogs
from components.DeleteBlogs import DeleteButton
from config import BASE_URL

app = Flask(__name__)

class UserBlogs(React.Component):
    def __init__(self):
        super().__init__()
        self.state = {
            'user': None
        }
        self.id = self.get_user_id()

    def get_user_id(self):
        # In a real Flask app, you'd use Flask-Login or similar for session management
        # This is a placeholder to mimic localStorage.getItem("userId")
        return request.cookies.get('userId')

    def send_request(self):
        try:
            response = requests.get(f"{BASE_URL}/api/blogs/user/{self.id}")
            response.raise_for_status()
            data = response.json()
            return data
        except requests.RequestException as e:
            print(f"Error: {e}")
            return None

    def component_did_mount(self):
        data = self.send_request()
        if data:
            self.setState({'user': data['user']})

    def handle_delete(self, blog_id):
        try:
            response = requests.delete(f"{BASE_URL}/api/blogs/{blog_id}")
            response.raise_for_status()
            data = self.send_request()
            if data:
                self.setState({'user': data['user']})
        except requests.RequestException as e:
            print(f"Error deleting blog: {e}")

    def render(self):
        user = self.state['user']
        if user and user['blogs']:
            return (
                <div className="container">
                    {user['blogs'].map((blog, index) => (
                        <div key={index} className="blog-container">
                            <Blogs
                                id={blog['_id']}
                                isUser={True}
                                title={blog['title']}
                                description={blog['description']}
                                imageURL={blog['image']}
                                userName={user['name']}
                            />
                            <img
                                className="blog-image"
                                src={blog['image']}
                                alt={blog['title']}
                            />
                            <DeleteButton blogId={blog['_id']} onDelete={self.handle_delete} />
                        </div>
                    ))}
                </div>
            )
        return None

@app.route('/api/user-blogs')
def user_blogs():
    user_blogs = UserBlogs()
    user_blogs.component_did_mount()
    return jsonify(user_blogs.state)

if __name__ == '__main__':
    app.run(debug=True)

File: mern-stack-converted-7f8d9e2a/react-frontend/source/components/UserBlogs.js
import React, { useEffect, useState } from 'react';
import axios from 'axios';
import Blog from './Blog';
import config from '../config';
import { makeStyles } from '@mui/styles';

const useStyles = makeStyles((theme) => ({
  container: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    margin: '20px auto',
    width: '80%',
  },
  blogContainer: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    padding: '20px',
    border: '1px solid #ccc',
    borderRadius: '10px',
    marginBottom: '20px',
    boxShadow: '0px 2px 4px rgba(0, 0, 0, 0.1)',
  },
}));

const UserBlogs = () => {
  const classes = useStyles();
  const [user, setUser] = useState();
  const id = localStorage.getItem('userId');

  const fetchUserBlogs = async () => {
    try {
      const response = await axios.get(`${config.BASE_URL}/api/blogs/user/${id}`);
      return response.data;
    } catch (error) {
      console.error('Error fetching user blogs:', error);
      return null;
    }
  };

  useEffect(() => {
    fetchUserBlogs().then((data) => {
      if (data) {
        setUser(data.user);
      }
    });
  }, []);

  const handleDelete = async (blogId) => {
    try {
      await axios.delete(`${config.BASE_URL}/api/blogs/${blogId}`);
      const updatedData = await fetchUserBlogs();
      if (updatedData) {
        setUser(updatedData.user);
      }
    } catch (error) {
      console.error('Error deleting blog:', error);
    }
  };

  return (
    <div className={classes.container}>
      {user &&
        user.blogs &&
        user.blogs.map((blog) => (
          <div key={blog._id} className={classes.blogContainer}>
            <Blog
              id={blog._id}
              isUser={true}
              title={blog.title}
              description={blog.description}
              imageURL={blog.image}
              userName={user.name}
            />
            <img
              className={classes.blogImage}
              src={blog.image}
              alt={blog.title}
            />
            <button onClick={() => handleDelete(blog._id)}>Delete</button>
          </div>
        ))}
    </div>
  );
};

export default UserBlogs;